server/src/main/java/org/apache/iotdb/db/writelog/node/ExclusiveWriteLogNode.java

```diff
@@ -171,6 +178,26 @@ public void close() {
     }
   }
 
+  @Override
+  public void release() {
+    ThreadUtils.stopThreadPool(FLUSH_BUFFER_THREAD_POOL, WAL_FLUSH);
+    lock.lock();
+    try {
+      if (this.logBufferWorking != null && this.logBufferWorking instanceof MappedByteBuffer) {
+        MmapUtil.clean((MappedByteBuffer) this.logBufferFlushing);
+      }
+      if (this.logBufferIdle != null && this.logBufferIdle instanceof MappedByteBuffer) {
+        MmapUtil.clean((MappedByteBuffer) this.logBufferIdle);
+      }
+      if (this.logBufferFlushing != null && this.logBufferFlushing instanceof MappedByteBuffer) {
+        MmapUtil.clean((MappedByteBuffer) this.logBufferFlushing);
+      }
```

Comment:
Excellent work! But ByteBuffers here are controlled by VirtualStorageGroupProcessor, whose releaseWalDirectByteBufferPool method also frees these ByteBuffers.
```
public void releaseWalDirectByteBufferPool() {
  synchronized (walByteBufferPool) {
    while (!walByteBufferPool.isEmpty()) {
      MmapUtil.clean((MappedByteBuffer) walByteBufferPool.removeFirst());
      currentWalPoolSize--;
    }
  }
}
```

Reply:
The buffers will be given back to `walByteBufferPool` in
https://github.com/ericpai/iotdb/blob/master/server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java#L1243-L1246. But you are right, the buffers won't be controlled by `walByteBufferPool` unless the corresponding tsfile is sealed.